hitl/streaming.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/python3
+"""
+Author: Vertigo Designs, Ryan Summers
+
+Description: Implements HITL testing of Stabilizer data livestream capabilities.
+"""
+import asyncio
+import sys
+import time
+import argparse
+import socket
+import struct
+import logging
+
+from miniconf import Miniconf
+
+def _get_ip(broker):
+    """ Get the IP of the local device.
+
+    Args:
+        broker: The broker IP of the test. Used to select an interface to get the IP of.
+
+    Returns:
+        The IP as an array of integers.
+    """
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        sock.connect((broker, 1883))
+        address = sock.getsockname()[0]
+    finally:
+        sock.close()
+
+    return list(map(int, address.split('.')))
+
+
+def sequence_delta(previous_sequence, next_sequence):
+    """ Check the number of items between two sequence numbers. """
+    if previous_sequence is None:
+        return 0
+
+    delta = next_sequence - (previous_sequence + 1)
+    return delta & 0xFFFFFFFF
+
+
+class StabilizerStream:
```

Comment:
Should this be moved into a common `stabilizer` python package that we place somewhere else? Then, do some `from stabilizer import StabilizerStream`?

Reply:
I'd leave it for now. Pretty sure we'll refactor/port/rewrite it when writing a web app or porting the stream decoding lib to rust.

Reply:
But in light of the upcoming CLI tool for the iir filters, yes, make this `/py/stream.py` and add `/py/__init__.py`. The CLI tool would also go there.

Right: `streaming_throughput` and this hitls test should both use a common lib for stream decoding.